lua/orgmode/config/init.lua

```diff
@@ -42,6 +42,9 @@ function Config:extend(opts)
   self.opts = vim.tbl_deep_extend('force', self.opts, opts)
   if self.org_startup_indented then
     self.org_adapt_indentation = not self.org_indent_mode_turns_off_org_adapt_indentation
+    if self.org_indent_mode_turns_on_hiding_stars then
+      self.org_hide_leading_stars = true
```

Comment:
I'm not 100% sure how Emacs does it, but we might need to set this to true/false depending on the `org_indent_mode` variable.

Reply:
Yes, I confirmed that's the case. It hides the stars only if `org_indent_mode` is enabled.

Let's add the check here  https://github.com/nvim-orgmode/orgmode/blob/7317cc97c45521d61104362adb30bcd299d1060c/lua/orgmode/colors/highlighter/stars.lua?plain=1#L18. Check for `org_indent_mode_turns_on_hiding_stars` and `vim.b[bufnr]org_indent_mode`. Depending on those proceed with hiding the stars.

Reply:
You beat me to the punch (I had a whole wall of text confirming the case lol). There is an issue though with just doing the check and calling it a day.

`org_indent_mode_turns_on_hiding_stars` in Emacs can be set on a buffer basis. Meaning that isn't *just* a global variable like we have right now. 

There's no way to toggle `org_indent_mode_turns_on_hiding_stars` in this PR for a single buffer (or at all for that matter) like you can in Emacs (granted I'm abusing Emacs `eval-expression` to do that).

With the above in mind, do you want me to migrate that to a `vim.b` variable similar to`vim.b.org_indent_mode` and make it toggleable?

Reply:
> With the above in mind, do you want me to migrate that to a vim.b variable similar tovim.b.org_indent_mode and make it toggleable or is it not worth it?

I don't think it's worth it. Let's just rely on the `vim.b.org_indent_mode` and the configuration for now, and we can add the improvement later if necessary.

Reply:
> I don't think it's worth it. Let's just rely on the vim.b.org_indent_mode and the configuration for now, and we can add the improvement later if necessary.

Ok sounds good. I can already see a small bug coming though so I'll get a PR out later this week to fix it.

Since `org_hide_leading_stars` is a "global" (meaning defined in `config`) variable in terms of the Orgmode plugin, reacting to `vim.b.org_indent_mode` will change the leading stars setting for other org buffers as well because they all look at `config.org_hide_leading_stars`.

Something to keep in mind.

Reply:
Done as part of the latest force push.